### PR TITLE
DDF-UI-112 allow intersects query for line

### LINKED ISSUE #112: Allow Line geo to have no buffer
Currently, we force the Line geo to have a buffer. User should be able to search with a Line without having any buffer. If the Line does have a buffer, it should be at least 1 meter. This reflects how we treat Polygons.

### PR DESCRIPTION
**Pull request by cassandrabailey293**
UI PORTION OF the Forward-port of https://github.com/codice/ddf/pull/5889

**This pr must be built on top of this DDF PR: https://github.com/codice/ddf/pull/5914

__________________________________________________________________________________
#### What does this PR do?
Allows a line to have a zero-buffer. Solves https://github.com/codice/ddf-ui/issues/112

#### Who is reviewing it? 

@hayleynorton @andrewzimmer @zta6 
#### Select relevant component teams: 

@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining 

#### How should this be tested?

- Advanced search - anyGeo
- Select Line
- Draw a line and verify that it doesn't automatically change the buffer width to 1
- Verify that you can search with nothing (and/or 0) in the buffer width input, and that the outgoing cql uses INTERSECTS instead of DWITHIN
- Verify that searching with a buffer width greater than 1 meter is allowed and that the outgoing cql uses DWITHIN
- Verify that searching with a buffer width of greater than 0 but less than 1 meter displays an inline error message "Buffer width must be 0 or at least 1 meter".  similar message should display when clicking search (the search should be prevented). Error message should change based on which units you have selected
- Verify no regression with Line searches

#### What are the relevant tickets?
Fixes:  https://github.com/codice/ddf-ui/issues/112

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js [hunk 1]
```diff
@@ -237,19 +234,6 @@ module.exports = Marionette.LayoutView.extend({
     )
     this.handleCurrentQuery()
 
-    // listener for when the measurement state changes in map model
-    this.listenTo(
```
Comment (andrewzimmer) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js]: why are these listeners being removed?

Comment (cassandrabailey293) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js]: i think this is actually an issue with the cherry-picking, it seems that my method of doing this overwrites changes in ddf-ui. I'll tag this pr with WIP for now so i can address these

**Review by andrewzimmer — APPROVED** [review]

**Review by bdeining — APPROVED** [review]

**Review by zta6 — APPROVED** [review]

**Review by hayleynorton — APPROVED** [review]

**Review by mojogitoverhere — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js [hunk 2]
```diff
@@ -604,6 +604,8 @@ module.exports = Marionette.LayoutView.extend({
               color,
             })
             this.map.showCircleShape(locationModel)
+          } else if (CQLUtils.isLineFilter(value)) {
+            this.handleFIlterAsLine(filter, color)
```
Comment (zta6) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js]: ✏️ Capital i in Filter

**Comment by andrewzimmer** [comment]
changes hero'd here https://github.com/codice/ddf/pull/5889#issuecomment-600350051

**Pull request merged**